packages/patrol_cli/lib/src/compatibility_checker/compatibility_checker.dart

```diff
@@ -86,10 +87,75 @@ class CompatibilityChecker {
     final isCompatible = areVersionsCompatible(cliVersion, patrolVersion);
 
     if (!isCompatible) {
-      throwToolExit(
-        'Patrol version $patrolVersion defined in the project is not compatible with patrol_cli version $cliVersion\n'
-        'Please upgrade both "patrol_cli" and "patrol" dependency in project to the latest versions.',
-      );
+      // Find the maximum compatible CLI version for this patrol version
+      final maxCliVersion = getMaxCompatibleCliVersion(patrolVersion);
+
+      if (maxCliVersion != null) {
+        throwToolExit(
+          'Patrol version $patrolVersion defined in the project is not compatible with patrol_cli version $cliVersion\n'
+          'To resolve this issue, you can:\n'
+          '1. Downgrade patrol_cli to a compatible version by running: \n'
+          '   ${lightCyan.wrap('dart pub global activate patrol_cli $maxCliVersion')}\n'
+          '2. Or upgrade both "patrol_cli" and "patrol" dependency in project to the latest versions.\n\n'
+          'Check the compatibility table at: ${lightCyan.wrap('https://patrol.leancode.co/documentation/compatibility-table')}',
+        );
+      } else {
+        throwToolExit(
+          'Patrol version $patrolVersion defined in the project is not compatible with patrol_cli version $cliVersion\n'
+          'Please upgrade both "patrol_cli" and "patrol" dependency in project to the latest versions.\n\n'
+          'Check the compatibility table at: ${lightCyan.wrap('https://patrol.leancode.co/documentation/compatibility-table')}',
+        );
+      }
+    }
+  }
+
+  /// Checks version compatibility and displays a warning if incompatible
+  Future<void> checkVersionsCompatibilityWithWarning({
+    required String? patrolVersion,
+  }) async {
+    if (patrolVersion == null) {
+      return;
+    }
+
+    try {
+      final cliVersion = Version.parse(constants.version);
+      final packageVersion = Version.parse(patrolVersion);
+
+      final isCompatible = areVersionsCompatible(cliVersion, packageVersion);
+      if (!isCompatible) {
+        // Find the maximum compatible CLI version for this patrol version
+        final maxCliVersion = getMaxCompatibleCliVersion(packageVersion);
+
+        if (maxCliVersion != null) {
+          _logger.warn(
+            '''
+Patrol version $packageVersion defined in your project is not compatible with patrol_cli version $cliVersion.
+This will cause issues when running 'patrol test'.
+
+To resolve this issue, you can:
+1. Downgrade patrol_cli to a compatible version by running: 
+   ${lightCyan.wrap('dart pub global activate patrol_cli $maxCliVersion')}
+   
+2. Or upgrade both "patrol_cli" and "patrol" dependencies to the latest versions.
+
+Check the compatibility table at: ${lightCyan.wrap('https://patrol.leancode.co/documentation/compatibility-table')}
+''',
```

Comment:
Maybe we could extract error/warning message to a common method to have something like:
```dart
String _incompatibilityMessage(
  String packageVersion,
  String? additionalInfo,
  String cliVersion,
  String? resolveSteps,
) {
  return '''
Patrol version $packageVersion defined in your project is not compatible with patrol_cli version $cliVersion.
$additionalInfo

${resolveSteps != null ? 'To resolve this issue, you can:\n$resolveSteps' : ''}

Check the compatibility table at: ${lightCyan.wrap('https://patrol.leancode.co/documentation/compatibility-table')}
''';
}
```
This way we will prevent duplicated parts of text